paymaster/build_txn.go

```diff
@@ -0,0 +1,395 @@
+package paymaster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/starknet.go/client/rpcerr"
+	"github.com/NethermindEth/starknet.go/typedata"
+)
+
+// BuildTransaction receives the transaction the user wants to execute. Returns the typed
+// data along with the estimated gas cost and the maximum gas cost suggested to ensure execution
+//
+// Parameters:
+//   - ctx: The context.Context object for controlling the function call
+//   - request: The BuildTransactionRequest containing the transaction and parameters
+//
+// Returns:
+//   - *BuildTransactionResponse: The response containing typed data and fee estimate
+//   - error: An error if the request fails
+func (p *Paymaster) BuildTransaction(
+	ctx context.Context,
+	request BuildTransactionRequest,
```

Comment:
Pass `request` by reference, please. There is no need to copy the entire value every time this function is called.